codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/SyncClientInterface.java

```diff
@@ -135,35 +139,62 @@ private MethodSpec serviceMetadata() {
             methods.add(simpleMethod(opModel));
         }
 
-        methods.add(operationMethodSignature(model, opModel)
+        methods.add(operationMethodSignature(model, opModel, opModel.getMethodName())
                             .addModifiers(Modifier.DEFAULT)
                             .addStatement("throw new $T()", UnsupportedOperationException.class)
                             .build());
 
         methods.addAll(streamingSimpleMethods(opModel));
 
+        // Add extra methods for paginated operations
+        if (opModel.isPaginated()) {
```

Comment:
Can we move all specific logic to other methods?

Basically end up with:

methods.addAll(paginatedSimpleMethods());
methods.addAll(paginatedMethods());

Reply:
I don't know if we want to have separate methods that go through all operations agains to find the paginated operations.

I moved the new code to a new method that takes in an operation model and return the method specs if its paginated operation.
List<MethodSpec> paginatedMethods(OperationModel opModel)

the call will end up with 
 methods.addAll(paginatedMethods(opModel));